tests/integration_tests/performance/test_snapshot_ab.py

```diff
@@ -0,0 +1,181 @@
+                jailer_binary_path=binaries / "jailer",
+                monitor_memory=False,
+            )
+            microvm.spawn()
+            microvm.restore_from_snapshot(snapshot, resume=True)
+
+            # Check if guest still runs commands.
+            exit_code, _, _ = microvm.ssh.run("true")
+            assert exit_code == 0
+
+            value = 0
+            # Parse all metric data points in search of load_snapshot time.
+            microvm.flush_metrics()
+            metrics = microvm.get_all_metrics()
+            for data_point in metrics:
+                cur_value = data_point["latencies_us"]["load_snapshot"]
+                if cur_value > 0:
+                    value = cur_value / USEC_IN_MSEC
+                    break
+            assert value > 0
+            values.append(value)
+            microvm.kill()
+            shutil.rmtree(Path(microvm.chroot()))
+
+        snapshot.delete()
+        return values
+
+
+@pytest.mark.nonci
+@pytest.mark.parametrize(
+    "test_setup",
+    [
+        SnapshotRestoreTest(mem=128, vcpus=1),
+        SnapshotRestoreTest(mem=1024, vcpus=1),
+        SnapshotRestoreTest(mem=2048, vcpus=2),
+        SnapshotRestoreTest(mem=4096, vcpus=3),
+        SnapshotRestoreTest(mem=6144, vcpus=4),
+        SnapshotRestoreTest(mem=8192, vcpus=5),
+        SnapshotRestoreTest(mem=10240, vcpus=6),
+        SnapshotRestoreTest(mem=12288, vcpus=7),
+        SnapshotRestoreTest(all_devices=True),
+    ],
+    ids=lambda x: x.id,
+)
+def test_restore_latency(microvm_factory, rootfs, guest_kernel_linux_4_14, test_setup):
+    """
+    Restores snapshots with vcpu/memory configuration, roughly scaling according to mem = (vcpus - 1) * 2048MB,
+    which resembles firecracker production setups. Also contains a test case for restoring a snapshot will all devices
+    attached to it.
+
+    We only test a single guest kernel, as the guest kernel does not "participate" in snapshot restore.
+    """
+    a_revision = os.environ.get("PERFORMANCE_TEST_REVISION_A")
+    b_revision = os.environ.get("PERFORMANCE_TEST_REVISION_B")
+
+    ah = test_setup.sample_latency(
+        microvm_factory,
+        guest_kernel_linux_4_14,
+        rootfs,
+        FC_WORKSPACE_DIR / "build" / a_revision,
+    )
+    be = test_setup.sample_latency(
+        microvm_factory,
+        guest_kernel_linux_4_14,
+        rootfs,
+        FC_WORKSPACE_DIR / "build" / b_revision,
+    )
+
+    result = check_regression(ah, be, larger_is_better=False)
+
+    assert (
+        result.pvalue > SIGNIFICANCE_THRESHOLD
+        or abs(result.statistic)
+        < abs(statistics.mean(ah)) * RELATIVE_STRENGTH_THRESHOLD_MS
+    ), f"A/B-testing shows a regression of {result.statistic:.2f}ms with p={result.pvalue}. This means that observing a regression of this magnitude or worse, assuming that performance characteristics did not change across the tested commits, has a probability of {result.pvalue:.2%}%"
```

Comment:
I think then you don't need the % outside.

Reply:
whoops, true!